src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorToolbar.ts

```diff
@@ -645,15 +645,17 @@ function actionOverflowHelper(initialPrimaryActions: IActionModel[], initialSeco
 		}
 	}
 
-	for (let i = (renderActions.length - 1); i !== 0; i--) {
-		const temp = renderActions[i];
-		if (temp.size === 0) {
-			continue;
-		}
-		if (temp.action instanceof Separator) {
-			renderActions.splice(i, 1);
+	if (renderActions.length) {
+		for (let i = (renderActions.length - 1); i !== 0; i--) {
```

Comment:
I think the reason we ran into this bug in the first place is we are checking `!== 0`, instead we should check `> 0`

Reply:
yeah good point